src/Maestro/SubscriptionActorService/LocalGit.cs

```diff
@@ -42,17 +44,26 @@ public LocalGit(TemporaryFiles tempFiles, IConfiguration configuration, Operatio
         public async Task<string> GetPathToLocalGitAsync()
         {
             // Determine whether we need to do any downloading at all.
-            if (!string.IsNullOrEmpty(_gitExecutable))
+            if (!string.IsNullOrEmpty(_gitExecutable) && File.Exists(_gitExecutable))
             {
-                return _gitExecutable;
+                // We should also mke sure that the git executable that exists runs properly
+                try
+                {
+                    LocalHelpers.ExecuteGitCommand(_gitExecutable, "--version", _logger, Environment.CurrentDirectory);
+                    return _gitExecutable;
+                }
+                catch (DarcException)
```

Comment:
So DarcException weirdly only gets throw if the output isn't returned.  That seems like a weirdly narrow determination that it's not fine.  Is there something in the "--version" output we could look at to make sure it behaved correctly?  So that we "fail" more often and get a "fresh" one?

Reply:
"git version --build-options" has some stuff that looks pretty parsable... like it should contain "built from commit" or something. :-)

Reply:
If ExecuteGitCommand returned the resulting string, we could probably check that the output contained "git version" in it. Right now, ExecuteGitCommand is a void function.

If there is an exception in running the git command, then my understanding is that ExecuteCommand will return null, and the DarcException will be thrown. I have not actually tested to see what happens if we have some sort of corrupted git.exe (honestly, not sure the best way to do that), but I have tested to see if, between checking that the file exists and running the git command, the file is deleted, and in that case, this does the right thing (catches the darc exception, and downloads a new git).

Reply:
Interestingly, my local git on my mac does not have a commit version attached to it. But that shouldn't matter since it's windows only.

Reply:
ExecuteGitCommand is... void? Ouch... uhm....uh.... it just... can fail horribly and there is NO feedback at all?  I don't have a great answer then. :-(

Reply:
Process stuff almost never throws.  I think the only time it would throw would be if "git.exe" didn't exist... but a simple File.Exists check could tell you that. :-)

Reply:
Ok, I've changed things around, making my own function for validating the git installation, using ExecuteCommand, not ExecuteGitCommand which is just a weird wrapper thingy that doesn't do any checking of the resulting output and is kind of redundant. @ChadNedzlek PTAL.